src/RollbarClient.mts

```diff
@@ -0,0 +1,49 @@
+// Internal Imports
+import type { IConfigurationOptions, TSubmitterParameters } from './types.mjs';
+import type { RollbarClientSubmitter } from './RollbarClientSubmitter.mjs';
+
+// Class Definition
+class RollbarClient {
+  configuration: IConfigurationOptions;
+  submitter?: RollbarClientSubmitter;
+
+  constructor(configurationOptions: IConfigurationOptions) {
+    this.configuration = configurationOptions;
+  }
+
+  initializeEventListeners() {
+    if (this.configuration.onUnhandledError !== false) {
+      window.addEventListener('error', this.configuration.onUnhandledError ?? this.onErrorDefault);
+    }
+    if (this.configuration.onUnhandledPromiseRejection !== false) {
+      window.addEventListener(
+        'unhandledrejection',
+        this.configuration.onUnhandledPromiseRejection ?? this.onUnhandledRejectionDefault,
+      );
+    }
+  }
+
+  async log(...parameters: TSubmitterParameters) {
+    if (!this.submitter) {
+      // eslint-disable-next-line import/no-unresolved -- false positive of an unimportable package
+      const { RollbarClientSubmitter } = await import('./RollbarClientSubmitter.mjs');
+      this.submitter = new RollbarClientSubmitter(this.configuration);
+    }
+    this.submitter.report(...parameters);
+  }
+
+  onErrorDefault = (errorEvent: ErrorEvent) => {
+    this.log('warning', 'Unhandled error occurred', errorEvent.error as Error).catch(() => {});
```

Comment:
Because `this.log()` is an async function and ESLint wants the promise handled. Since we don't care about the return value (e.g. `sendBeacon()`), I added `.catch()` to handle the error case and stop ESLint complaining.

Reply:
I see.
I am not a big fan in general of swallowing errors without a warning, the error this would report could eventually be interesting during the development phase (I agree that in production, if it's not going to go to your error reporter, it's relatively useless, except perhaps if the developer is the one currently checking the problem 😅).

Maybe there could be an option you can pass to the library something like `silentlyIgnoreNetworkReportingErrors` that one could set accordingly to their wishes?
(I think it wouldn't be useful a lot, but at least it would make the behavior a little more explicit/configurable)

Reply:
I've thought about this more, and I've changed this to await the log call so that any exceptions that get thrown will occur at the call site; thank you.

The reason I'm not handling the exceptions is that the only possible exception would come from a failing `fetch` call, and this is an error reporting library wherein the `fetch` call is the main feature, so now if that happens the user will see the exception thrown in their console and know where it came from whereas before it would have been swallowed like you said.

Reply:
Yeah makes sense to me!

Reply:
A further adaptation is that I removed the `await` to keep the event listeners non-`async` and instead I'm returning the `this.log()` calls which will have the same effect. There are edge case issues with making event listeners async, so I'm being a bit defensive here because I don't know how end users will use this library. Better safe than sorry.